WireDomain/Tests/WireDomainTests/Synchronization/SyncManagerTests.swift

```diff
@@ -349,62 +384,111 @@ final class SyncManagerTests: XCTestCase {
-        conversationID: conversationID1,
-        senderID: aliceID,
-        isTyping: true
-    )))
-
-    static let event3 = UpdateEvent.conversation(.delete(ConversationDeleteEvent(
-        conversationID: conversationID1,
-        senderID: aliceID,
-        timestamp: .now
-    )))
-
-    static let event4 = UpdateEvent.conversation(.rename(ConversationRenameEvent(
-        conversationID: conversationID2,
-        senderID: aliceID,
-        timestamp: .now,
-        newName: "Foo"
-    )))
-
-    static let event5 = UpdateEvent.conversation(.rename(ConversationRenameEvent(
-        conversationID: conversationID2,
-        senderID: aliceID,
-        timestamp: .now,
-        newName: "Bar"
-    )))
-
-    static func makeEnvelope(
-        with event: UpdateEvent,
-        isTransient: Bool = false
-    ) -> UpdateEventEnvelope {
-        .init(
-            id: UUID(),
-            events: [event],
-            isTransient: isTransient
-        )
+        let user = await context.perform { [self] in
+            modelHelper.createUser(in: context)
+        }
+
+        let (selfUser, selfUserID) = await context.perform { [self] in
+            let selfUser = modelHelper.createSelfUser(in: context)
+            let selfUserID: UUID = selfUser.remoteIdentifier
+
+            return (selfUser, selfUserID)
+        }
+
+        let conversation = await context.perform { [self] in
+            modelHelper.createGroupConversation(in: context)
+        }
+
+        updateEventsRepository.pullLastEventID_MockMethod = {}
+        teamRepository.pullSelfTeam_MockMethod = {}
+        teamRepository.pullSelfTeamRoles_MockMethod = {}
+        teamRepository.pullSelfTeamMembers_MockMethod = {}
+        connectionsRepository.pullConnections_MockMethod = {}
+        conversationsRepository.pullConversations_MockMethod = {}
+        conversationsRepository.pullMLSOneToOneConversationUserIDDomain_MockValue = UUID().uuidString
+        conversationsRepository.fetchMLSConversationWith_MockValue = conversation
+        userRepository.pullKnownUsers_MockMethod = {}
+        conversationLabelsRepository.pullConversationLabels_MockMethod = {}
+        featureConfigsRepository.pullFeatureConfigs_MockMethod = {}
+        userRepository.pullSelfUser_MockError = UserRepositoryError.failedToFetchUser(selfUserID) /// throws error
+        teamRepository.pullSelfLegalholdInfo_MockMethod = {}
+        pushSupportedProtocolsUseCase.invoke_MockMethod = {}
+        userRepository.fetchAllUserIDsWithOneOnOneConversation_MockMethod = { [] }
+        userRepository.fetchUserWithDomain_MockValue = user
+        userRepository.fetchSelfUser_MockValue = selfUser
+        mlsService.conversationExistsGroupID_MockValue = true
+        mlsService.establishGroupForWithRemovalKeys_MockValue = .MLS_128_DHKEMP256_AES128GCM_SHA256_P256
+        mlsService.joinGroupWith_MockMethod = { _ in }
+
+        do {
+            try await sut.performSlowSync()
```

Comment:
```suggestion
            try await sut.performSlowSync()
            XCTFail("this test should raise an error")
```

Reply:
done in https://github.com/wireapp/wire-ios/pull/1999/commits/e151fbc4913b20e97bb8214302a8f3558944479c